### PR TITLE
Warning for duplicated deps on project root level

### LINKED ISSUE #5741: Show warning when a package is defined twice in mix.exs
Currently, if you have the following in your `mix.exs`:

``` elixir
  defp deps do
    [{:ecto, "~> 2.1"},
     {:postgrex, "~> 0.13.0"},
     {:ecto, "~> 2.0"}]
  end
```

The last definition for ecto is ignored and `~> 2.1` is resolved. It could be useful if multiple definitions of the same dependency (in the same file) exist, to show a warning.

### PR DESCRIPTION
**Pull request by kelvinst**
This PR closes #5741

### CODE REVIEW

File: lib/mix/lib/mix/dep/converger.ex [hunk 1]
```diff
@@ -218,6 +218,10 @@ defmodule Mix.Dep.Converger do
     {acc, match} =
       Enum.map_reduce list, false, fn(other, match) ->
         %Mix.Dep{app: other_app, opts: other_opts} = other
+        if other_app == app and other.top_level and dep.top_level do
+          Mix.shell.error "warning: the dependency #{dep.app} is duplicated " <>
```
Comment (josevalim) [inline lib/mix/lib/mix/dep/converger.ex]: `#{inspect dep.app}`

**Comment by josevalim** [comment]
One comment and we should be good to go!

**Comment by kelvinst** [comment]
Done and squashed!

**Comment by jeroenvisser101** [comment]
Thanks! 🎉

**Comment by fishcakez** [comment]
Thank you :heart:

**Comment by kelvinst** [comment]
You are very welcome!!

**Pull request merged**